### PR TITLE
Refs #16416 -  Retrieve password_hash from OS

### PR DESCRIPTION
**Pull request by sean797**
Requires theforeman/foreman#3805

### CODE REVIEW

**Comment by sean797** [comment]
[test]

File: kickstart/provision.erb [hunk 1]
```diff
@@ -56,7 +56,7 @@ firewall --disable
 <% else -%>
 firewall --<%= os_major >= 6 ? 'service=' : '' %>ssh
 <% end -%>
-authconfig --useshadow --passalgo=sha256 --kickstart
+authconfig --useshadow --passalgo=<%= @host.operatingsystem.password_hash %> --kickstart
```
Comment (ekohl) [inline kickstart/provision.erb]: Maybe you can fall back to sha256 if it's not set? That way we stay compatible with older foreman versions.

**Comment by ekohl** [comment]
There's a typo in the commit message

**Comment by mmoll** [comment]
Also, tests need updating. :+1: otherwise

**Comment by mmoll** [comment]
merged as 519490c, thanks @sean797!

**Pull request closed without merging**